File: Sesion 7/Ejercicio/Ejercicio1 S6/Ejercicio1.py
#CLASE PADRE
class Vehiculo:
    def __init__(self, marca, modelo, anio, precio):
        self.marca = marca
        self.modelo = modelo
        self.anio = anio
        self.precio = precio

    def mostrarDatos(self):
        print(f"Marca: {self.marca}")
        print(f"Modelo: {self.modelo}")
        print(f"Año: {self.anio}")
        print(f"Precio: {self.precio} Soles.")

#CLASES HIJOS
class Automovil(Vehiculo):
    def __init__(self, marca, modelo, anio, precio, numPuertas, tipoCombustible):
        super().__init__(marca, modelo, anio, precio)
        self.numPuertas = numPuertas
        self.tipoCombustible = tipoCombustible

    def mostrarDatos(self):
        super().mostrarDatos()
        print(f"Número de puertas: {self.numPuertas}")
        print(f"Tipo de combustible: {self.tipoCombustible}")

class Motocicleta(Vehiculo):
    def __init__(self, marca, modelo, anio, precio, cilindrada):
        super().__init__(marca, modelo, anio, precio)
        self.cilindrada = cilindrada

    def mostrarDatos(self):
        super().mostrarDatos()
        print(f"Cilindrada: {self.cilindrada}")

# Main
auto1 = Automovil("Kia", "Rio", 2020, 60000, 4, "Gasolina")
moto1 = Motocicleta("Honda", "CRF450R", 2021, 45000, 450)

print("Datos del Carro:")
auto1.mostrarDatos()

print("\nDatos de la Moto:")
moto1.mostrarDatos()

File: Sesion 7/Ejercicio/Ejercicio1 S6/Ejercicio1.java
// CLASE PADRE
class Vehiculo {
    protected String marca, modelo;
    protected int anio;
    protected double precio;

    public Vehiculo(String m, String mod, int a, double p) {
        marca = m;
        modelo = mod;
        anio = a;
        precio = p;
    }

    public void mostrarDatos() {
        System.out.println("Marca: " + marca);
        System.out.println("Modelo: " + modelo);
        System.out.println("Año: " + anio);
        System.out.println("Precio: " + precio + " Soles.");
    }
}

// CLASES HIJOS
class Automovil extends Vehiculo {
    private int numPuertas;
    private String tipoCombustible;

    public Automovil(String m, String mod, int a, double p, int np, String tc) {
        super(m, mod, a, p);
        numPuertas = np;
        tipoCombustible = tc;
    }

    @Override
    public void mostrarDatos() {
        super.mostrarDatos();
        System.out.println("Número de puertas: " + numPuertas);
        System.out.println("Tipo de combustible: " + tipoCombustible);
    }
}

class Motocicleta extends Vehiculo {
    private int cilindrada;

    public Motocicleta(String m, String mod, int a, double p, int c) {
        super(m, mod, a, p);
        cilindrada = c;
    }

    @Override
    public void mostrarDatos() {
        super.mostrarDatos();
        System.out.println("Cilindrada: " + cilindrada);
    }
}

public class Main {
    public static void main(String[] args) {
        Automovil auto1 = new Automovil("Kia", "Rio", 2020, 60000, 4, "Gasolina");
        Motocicleta moto1 = new Motocicleta("Honda", "CRF450R", 2021, 45000, 450);

        System.out.println("Datos del Carro:");
        auto1.mostrarDatos();

        System.out.println("\nDatos de la Moto:");
        moto1.mostrarDatos();
    }
}